src/legend/Legend.component.js

```diff
@@ -77,14 +77,22 @@ class Legend extends Component {
         this.props.onItemsChange(newItems);
     };
 
-    updateItem = (newItems) => {
-        const modelToUpdate = legendItemStore.getState() && legendItemStore.getState().model;
-        const isNewLegendItem = newItems.every(model => model !== modelToUpdate);
-
-        return this.props.onItemsChange([].concat(
-            newItems,
-            isNewLegendItem ? modelToUpdate : [],
-        ));
+    updateItem = (existingItems) => {
+        let updatedItems = [];
+
+        const currentItem = legendItemStore.getState() && legendItemStore.getState().model;
+        const isNewItem = existingItems.every(item => item.id !== currentItem.id);
```

Comment:
I know this code was already existing, but this would be a little more efficient as `!existingItems.some(item => item.id === currentItem.id)` since it would short-circuit the iteration once a match was found.